### PR TITLE
Limit the automation-editor scaled-level tooltip to the grid

### LINKED ISSUE #3746: Automation points tooltips can show outside of the automation editor border & window.
Dragging or dragging & releasing an automation point shows a tooltip with its value. The issue is that this can happen outside the automation editor border & window.

Outside the automation editor border:
image
Outside the automation editor window:
image

The above tooltips showed after dragging an automation point and releasing (by releasing left mouse-button) outside the automation editor border & window.

I suggest setting limits on where automation points tooltips can show.
My suggested limit is the automation editor border, the one highlighted in yellow in this image:
snip_20170802184919

### PR DESCRIPTION
**Pull request by Sawuare**
Fixes #3746.

### CODE REVIEW

**Comment by zonkmachine** [comment]
Here is another issue that sounds `AutomationEditor::drawCross()` related: https://github.com/LMMS/lmms/issues/2714

File: src/gui/editors/AutomationEditor.cpp [hunk 1]
```diff
@@ -1076,23 +1076,36 @@ void AutomationEditor::mouseMoveEvent(QMouseEvent * mouseEvent )
...
-	tt_pos.ry() -= 64;
-	tt_pos.rx() += 32;
+	tt_pos.ry() -= 51;
+	tt_pos.rx() += 26;
+
 	float scaledLevel = m_pattern->firstObject()->scaledValue( level );
-	QToolTip::showText( tt_pos, QString::number( scaledLevel ), this );
+
+	// Limit the scaled-level tooltip to the grid
+	if(	mouse_pos.x() > LEFT_MARGIN &&
```
[the remark this answers is not in the document]
Comment (Sawuare) [reply src/gui/editors/AutomationEditor.cpp]: Mistake. Fixed in https://github.com/LMMS/lmms/pull/4308/commits/d1dfe325f9ecd33bd06627014e0ae305dc3191f2.

File: include/AutomationEditor.h [hunk 2]
```diff
@@ -173,6 +173,7 @@ protected slots:
 
 	// some constants...
 	static const int SCROLLBAR_SIZE = 12;
+	static const int LEFT_MARGIN = 63;
```
[the remark this answers is not in the document]
Comment (Sawuare) [reply include/AutomationEditor.h]: As in using `VALUES_WIDTH` instead of `LEFT_MARGIN` in `drawCross()`?

[the remark this answers is not in the document]
Comment (PhysSong) [reply include/AutomationEditor.h]: Yes, though it doesn't need to be simple replacing.

[the remark this answers is not in the document]
Comment (Sawuare) [reply include/AutomationEditor.h]: Done in 19b0342.

File: src/gui/editors/AutomationEditor.cpp [hunk 3]
```diff
@@ -1099,8 +1098,8 @@ inline void AutomationEditor::drawCross( QPainter & p )
 	float scaledLevel = m_pattern->firstObject()->scaledValue( level );
 
 	// Limit the scaled-level tooltip to the grid
-	if( mouse_pos.x() > LEFT_MARGIN &&
-		mouse_pos.x() < right_margin &&
+	if( mouse_pos.x() > VALUES_WIDTH - 1 && // left_margin
```
Comment (PhysSong) [inline src/gui/editors/AutomationEditor.cpp]: I think `VALUES_WIDTH` would be better, because it matches the limit area to where you can edit points.

**Comment by Sawuare** [comment]
> Here is another issue that sounds AutomationEditor::drawCross() related: #2714

It is. It can be fixed with similar checks, so I will do it here.

**Comment by Sawuare** [comment]
@zonkmachine After some thinking, I don't think d127517 is the best way to fix #2714. It feels clunky, and it's somewhat out of scope. I will try to fix it in a different way, but not here.
That being said, I will finish this tomorrow.

**Pull request merged**